h2o-core/src/main/java/water/api/Handler2.java

```diff
@@ -0,0 +1,26 @@
+package water.api;
+
+/**
+ * Common interface for <s>all</s> some REST endpoint handlers.
+ *
+ * @param <IS> input schema class
+ * @param <OS> output schema class
+ */
+public abstract class Handler2<IS extends Schema, OS extends Schema> extends Handler {
```

Comment:
So what could be a good name? How about `JollyHandler` or `BetterHandler` or `NextHandler` or `RealHandler` or `RestApiHandler`? It's hard to come up with a good name within a hierarchy which is screwed up (in the sense that there is already a class named `Handler`, and multiple classes which inherit from it, even though they should not).